pkgs/development/libraries/glibc/common.nix

```diff
@@ -96,6 +96,14 @@ stdenv.mkDerivation ({
          & https://github.com/NixOS/nixpkgs/pull/188492#issuecomment-1233802991
       */
       ./reenable_DT_HASH.patch
+
+      /* NVCC does not support ARM intrinsics. Since <math.h> is pulled in by almost
+         every HPC piece of software, without this patch CUDA compilation on ARM
+         is effectively broken.
+
+         See https://forums.developer.nvidia.com/t/nvcc-fails-to-build-with-arm-neon-instructions-cpp-vs-cu/248355/2.
+      */
+      ./0001-aarch64-math-vector.h-add-NVCC-include-guard.patch
     ]
     ++ lib.optional stdenv.hostPlatform.isMusl ./fix-rpc-types-musl-conflicts.patch
     ++ lib.optional stdenv.buildPlatform.isDarwin ./darwin-cross-build.patch;
```

Comment:
We could also wrap this in `optionals stdenv.isAarch64`, avoiding the rebuild for x86_64-linux. I like the current way more though, it's simpler

Reply:
That's a good idea; it reduces the number of rebuilds and we can see that there are similar platform-specific patches that aren't applied unconditionally.